components/UI/searchBar.tsx

```diff
@@ -1,68 +1,202 @@
+      setCurrentResult(result);
+    });
   }
 
-  function onEnter(ev: KeyboardEvent<HTMLDivElement>) {
-    if (ev.key === "Enter") {
-      search(typedValue);
-      ev.preventDefault();
+  async function getStatus(
+    name: string,
+    lastAccessed?: number
+  ): Promise<SearchResult> {
+    const isDomainValid = useIsValid(name);
+    if (isDomainValid !== true) {
+      return {
+        name,
+        error: true,
+        message: isDomainValid + " is not a valid caracter",
+        lastAccessed: lastAccessed ?? Date.now(),
+      };
+    } else {
+      const currentTimeStamp = new Date().getTime() / 1000;
+      const encoded = name
+        ? utils.encodeDomain(name).map((elem) => elem.toString())
+        : [];
+      return new Promise((resolve) => {
+        contract?.call("domain_to_expiry", [encoded]).then((res) => {
+          if (Number(res?.["expiry"]) < currentTimeStamp) {
+            resolve({
+              name,
+              error: false,
+              message: "Available",
+              lastAccessed: lastAccessed ?? Date.now(),
+            });
+          } else {
+            resolve({
+              name,
+              error: true,
+              message: "Unavailable",
+              lastAccessed: lastAccessed ?? Date.now(),
+            });
+          }
+        });
+      });
     }
   }
 
   function search(typedValue: string) {
     if (typeof isDomainValid === "boolean") {
       onChangeTypedValue?.(typedValue);
+      saveSearch(currentResult as SearchResult);
+      setCurrentResult(null);
+      setTypedValue("");
       router.push(`/search?domain=${typedValue}`);
     }
   }
 
-  return (
-    <TextField
-      fullWidth
-      className="z-[0]"
-      id="outlined-basic"
-      label={
-        isDomainValid != true
-          ? `"${isDomainValid}" is not a valid character`
-          : "Your Username"
+  function saveSearch(newResult: SearchResult) {
+    setSearchResults((prevResults) => {
+      const updatedResults = [...(prevResults || [])]; // Clone the previous results
+      const existingResult = updatedResults.find(
+        (result) => result.name === newResult.name
+      );
+
+      if (existingResult) {
+        existingResult.lastAccessed = Date.now();
+      } else {
+        newResult.lastAccessed = Date.now();
+        updatedResults.unshift(newResult);
       }
-      placeholder="Type your .stark domain here"
-      variant="outlined"
-      onChange={(e) => handleChange(e.target.value)}
-      onKeyPress={(ev) => onEnter(ev)}
-      InputProps={{
-        endAdornment: (
-          <InputAdornment
-            onClick={() => {
-              if (typedValue) search(typedValue);
-            }}
-            className="cursor-pointer"
-            position="end"
-          >
-            <SearchIcon />
-          </InputAdornment>
-        ),
-      }}
-      error={isDomainValid != true}
-    />
+
+      updatedResults.sort((a, b) => b.lastAccessed - a.lastAccessed);
```

Comment:
can you find more relevant names instead of `a` & `b` ? It'll make the code easier to understand